public/video-ui/src/components/YoutubeFurniture/index.js

```diff
@@ -0,0 +1,149 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { ManagedField, ManagedForm } from '../ManagedForm';
+import VideoUtils from '../../util/video';
+import TagTypes from '../../constants/TagTypes';
+import TagPicker from '../FormFields/TagPicker';
+import SelectBox from '../FormFields/SelectBox';
+import PrivacyStates from '../../constants/privacyStates';
+import {formNames} from "../../constants/formNames";
+import YouTubeKeywords from "../../constants/youTubeKeywords";
+import {getYouTubeTagCharCount} from "../../util/getYouTubeTagCharCount";
+import FieldNotification from "../../constants/FieldNotification";
+import KeywordsApi from '../../services/KeywordsApi';
+
+class YoutubeFurniture extends React.Component {
+  static propTypes = {
+    video: PropTypes.object.isRequired,
+    editable: PropTypes.bool.isRequired,
+    updateVideo: PropTypes.func.isRequired,
+    updateErrors: PropTypes.func.isRequired,
+    updateWarnings: PropTypes.func.isRequired
+  };
+
+  hasCategories = () => this.props.youtube.categories.length !== 0;
+  hasChannels = () => this.props.youtube.channels.length !== 0;
+
+  componentWillMount() {
+    if (!this.hasCategories()) {
+      this.props.youtubeActions.getCategories();
+    }
+    if (!this.hasChannels()) {
+      this.props.youtubeActions.getChannels();
+    }
+  }
+
+  validateYouTubeKeywords = youTubeKeywords => {
+    const charLimit = YouTubeKeywords.maxCharacters;
+    const numberOfChars = getYouTubeTagCharCount(youTubeKeywords);
+    const characterCountExceeded = numberOfChars > charLimit;
+
+    return characterCountExceeded
+      ? new FieldNotification(
+        'required',
+        `Maximum characters allowed in YouTube keywords is ${charLimit}.`,
+        FieldNotification.error
+      )
+      : null;
+  };
+
+  composerKeywordsToYouTube = () => {
+    const { video, updateVideo } = this.props;
+
+    return Promise.all(
+      video.keywords.map(keyword => KeywordsApi.composerTagToYouTube(keyword))
+    )
+      .then(youTubeKeywords => {
+        const oldTags = video.tags;
+        const keywordsToCopy = youTubeKeywords.reduce((tagsAdded, keyword) => {
+          const allAddedTags = oldTags.concat(tagsAdded);
+          if (keyword !== '' && allAddedTags.every(oldTag => oldTag !== keyword)) {
+            tagsAdded.push(keyword);
+          }
+          return tagsAdded;
+        }, []);
+        const newVideo = Object.assign({}, video, { tags: oldTags.concat(keywordsToCopy)});
+        updateVideo(newVideo);
+      });
+  };
+
+  render() {
+    const {
+      video,
+      editable,
+      updateVideo,
+      updateErrors,
+      updateWarnings,
+    } = this.props;
+
+    const { categories } = this.props.youtube;
+
+    const isYoutubeAtom = VideoUtils.isYoutube(video);
```

Comment:
Are these calls, e.g. getAvailableChannels, triggering side-effects? Are we sure we want to make them every render?